turtlesim/CMakeLists.txt

```diff
@@ -13,15 +13,16 @@ endif()
 find_package(ament_cmake REQUIRED)
 find_package(ament_index_cpp REQUIRED)
 find_package(geometry_msgs REQUIRED)
-find_package(Qt5 REQUIRED COMPONENTS Widgets)
+find_package(QT NAMES Qt5 Qt6 REQUIRED COMPONENTS Widgets Core)
+find_package(Qt${QT_VERSION_MAJOR} REQUIRED COMPONENTS Widgets Core)
 find_package(rcl_interfaces REQUIRED)
 find_package(rclcpp REQUIRED)
 find_package(rclcpp_action REQUIRED)
 find_package(rosidl_default_generators REQUIRED)
 find_package(std_msgs REQUIRED)
 find_package(std_srvs REQUIRED)
 
-include_directories(include ${Qt5Widgets_INCLUDE_DIRS})
+include_directories(include ${Qt${QT_VERSION_MAJOR}Widgets_INCLUDE_DIRS})
```

Comment:
While we are messing with this, I think we should actually move away from it.  In particular, we should use `target_include_directories`, similar to https://github.com/ros2/demos/blob/2614f4ad6b96c86ead4513c142f531348669f61f/demo_nodes_cpp/CMakeLists.txt#L48-L50